### DISCUSSION TITLE
[Doc] Earn More Featured Snippets for Pulsar Doc

### DISCUSSION THREAD

**Discussion by zuobiao-zhou** [apache/pulsar#20897] [Ideas]
## Motivation

Nowadays, an increasing number of users are searching for Pulsar content online, and the **search visibility of official documentation** has become increasingly important.

In Google search result, **featured snippets** [1] appear **at the top** where usually called “**Page 0**”. Featured snippets provide **a brief text that directly answers the searcher's query**.

<img width="320" alt="0号位置" src="https://github.com/apache/pulsar/assets/61108539/54e3aee3-36bf-4b95-aebb-4fef2b489fc4">

Compared to the **19.6% click-through rate** of **all natural search results** under Featured snippet, **Featured snippet** gets about **8.6% of the click-through rate**. As you can see, Featured snippets contribute significantly to the pageview of a website.

<img width="320" alt="0号位置点击率" src="https://github.com/apache/pulsar/assets/61108539/e4b1409d-91d8-4472-84da-d6aaee946d79">

Among all SEO [2] (Search Engine Optimization) strategies, **earning a featured snippet is the primary and most effective method**.

Therefore, I would like to propose making some small modifications to the Pulsar documentation to **earn more featured snippets**, thereby **improving pageview numbers**.

## 4 main types of featured snippets

There are 4 main types of featured snippets in Google: **definition**, **table**, **unordered list**, and **ordered list**.

- **definition**
	This is a text snippet designed to provide searchers with **a direct and concise definition or description**. Google typically uses a definition box to answer searches for "**what is …**"

![image](https://github.com/apache/pulsar/assets/61108539/bc77810a-fdf2-447e-9363-f6ca8087ced5)

- **table**
	Google extracts data from webpages and displays it as a table.

![image](https://github.com/apache/pulsar/assets/61108539/fc474e14-adaa-4580-b8d2-ed3915a634ec)

- **unordered list**
	Google uses an unordered list to display items that do not need to be arranged in a specific order.

![image](https://github.com/apache/pulsar/assets/61108539/1887052a-4286-426d-afba-42097149ee2f)

- **ordered list**
	This is a list displayed in a specific order. Google typically uses it to respond to search queries that require sequential steps.

![image](https://github.com/apache/pulsar/assets/61108539/2890d710-8ec7-4415-be12-6d5663d9c43a)

## How to earn more featured snippets for Pulsar doc

The Pulsar doc already has a certain level of authority. By standardizing the format to make the content easily understandable by search engines, there is a high probability of generating featured snippets.

The types of featured snippets that Pulsar doc can generate typically correspond to definition, unordered list and ordered list.

By drawing on some SEO best practices and studying how other websites create featured snippets, I have summarized the techniques for generating featured snippets as follows:

- definition
	Google typically uses definition boxes to answer searches for "what is (are) X". The average length of these snippets ranges from 40 to 60 words.

	To generate this type of featured snippet, we need to explicitly write sentences in the document using the structure "X is (are) …" and emphasize the defining part.

	For example, we should use "**Producer is** a process that publishes messages to a Pulsar topic" here

![image](https://github.com/apache/pulsar/assets/61108539/d0e0649e-3e24-405a-80ca-004e001e94f1)

- unordered list
	This is a list displayed of items that do not need to be arranged in a specific order.

	To generate this type of featured snippet, Place each item on a separate line and precede it with a bullet point for the list.

	For example, we should adopt the following format.

![image](https://github.com/apache/pulsar/assets/61108539/942f732b-eeb0-4bea-9f82-17f9a4ff80f9)

- ordered list
	This is a list displayed in a specific order, corresponds to the "How to ..." type of query.

	To generate this type of featured snippet, we should use a series of summary-style subheadings in the body of the content and explicitly define each step by adding the relevant phrases like "Step 1 Step 2 Step 3 ..." or "1. 2. 3. …" before them.

	For example, we should make the following modifications.

<img width="889" alt="图片1" src="https://github.com/apache/pulsar/assets/61108539/20919fcd-8c65-46fc-a062-ff9ce2d170bd">

## Schedule

In the next two months, I will do SEO for Pulsar Doc [3], which includes optimizing the content to earn featured snippets and adding meta descriptions [4] and alt tags [5].

[1] https://developers.google.com/search/docs/appearance/featured-snippets
[2] https://developers.google.com/search/docs/fundamentals/seo-starter-guide
[3] https://pulsar.apache.org/docs/next/
[4] https://developers.google.com/search/docs/appearance/snippet
[5] https://developers.google.com/search/docs/appearance/google-images

**Comment by Anonymitaet** [comment]
This is the OSPP project -- [Optimizing the findability of Pulsar documentation with SEO strategies](https://summer-ospp.ac.cn/org/prodetail/23e980570?list=org&navpage=org), aiming to improve the findability and searchability of content on the Pulsar documentation website by applying effective SEO strategies.

For a detailed implementation plan, see [here](https://docs.google.com/spreadsheets/d/1usdvJDEAV9Kxx5tue8O_amoAfnxFcQ0mMfwCZ2323nU/edit#gid=0).
